src/container.ts

```diff
@@ -889,14 +885,118 @@ export interface MemoryResources {
   readonly limit: container;
 }
 
-function renderEnv(env: { [name: string]: EnvValue }): k8s.EnvVar[] {
-  const result = new Array<k8s.EnvVar>();
-  for (const [name, v] of Object.entries(env)) {
-    result.push({
-      name,
-      value: v.value,
-      valueFrom: v.valueFrom,
-    });
+/**
+ * A collection of env variables defined in other resources.
+ */
+export class EnvFrom {
+
+  constructor(
+    private readonly configMap?: configmap.IConfigMap,
+    private readonly prefix?: string,
+    private readonly sec?: secret.ISecret) {};
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.EnvFromSource {
+    return {
+      configMapRef: this.configMap ? {
+        name: this.configMap.name,
+      } : undefined,
+      secretRef: this.sec ? {
+        name: this.sec.name,
+      } : undefined,
+      prefix: this.prefix,
+    };
   }
-  return result;
+
 }
+
+/**
+ * Container environment.
+ */
```

Comment:
optional - "Container environment" sounds broad to me, like it could refer to any aspect of the host environment.
```suggestion
/**
 * Container environment variables.
 */
```

I was also thinking about whether it would be good to name these `ContainerEnv` and `ContainerEnvFrom` since these seem more like helper classes specific to container use cases. But all of kubernetes is containers, so I'm probably overthinking it

Reply:
> But all of kubernetes is containers, so I'm probably overthinking it

Yeah, I think its fine as is. Adding the `Container` prefix just makes the usage more verbose for no real benefit - k8s also names it like so.